staging/maven-plugin/src/main/java/org/sonatype/nexus/maven/staging/deploy/strategy/StagingDeployStrategy.java

```diff
@@ -73,14 +78,19 @@ public void deployPerModule(final DeployPerModuleRequest request)
       final String profileId =
           selectStagingProfile(request.getParameters(), remoteNexus,
               request.getDeployableArtifacts().get(0).getArtifact());
+
+      // Store local nexus staging repositories in folder stagingDirectoryRoot/MD5OfNexusUrl/profileId
+      // Because multiple nexus instances might contain the same profileId
+      String stagingFolderName = generateMD5String(request.getParameters().getNexusUrl()) + File.separator + profileId;
       final File stagingDirectory =
-          getStagingDirectory(request.getParameters().getStagingDirectoryRoot(), profileId);
+          getStagingDirectory(request.getParameters().getStagingDirectoryRoot(), stagingFolderName);
```

Comment:
Install to `<local staging root>/<md5 of nexus url>/<profileId>` instead of `<local staging root>/<profile id>`